network-api/networkapi/buyersguide/tests.py

```diff
@@ -440,18 +460,455 @@ def name_change_changes_slug(self):
+                'track_record_choices', 'company_track_record', 'track_record_is_bad',
+                'track_record_details', 'offline_capable', 'offline_use_description',
+                'uses_ai', 'ai_uses_personal_data', 'ai_is_transparent', 'ai_helptext'
+            ]
+            # For every field, compare it to the old field in the Product object
+            for field in specific_fields:
+                self.assertEqual(
+                    getattr(product, field),
+                    getattr(product_page, field),
+                    f"{field} did not match on product: {product} (id:{product.id})",
+                )
+
+    def test_migration(self):
+        """
+        All tests go in here to ensure they are executed in the proper order
+        """
+        self.before_migrate()
+        self.migrate()
+        self.after_migrate()
+        self.check_buyersguide_exists()
+        self.check_general_products_match()
+        self.check_software_products_match()
+
+
+class TestProductPage(BuyersGuideTestMixin):
+
+    def setUp(self):
+        super().setUp()
+        if not hasattr(self.product_page.votes, 'get_votes'):
+            votes = ProductPageVotes()
+            votes.save()
+            self.product_page.votes = votes
+            self.product_page.save()
+
+    def test_get_votes(self):
+        # Votes should be empty at this point.
+        votes = self.product_page.votes.get_votes()
+        self.assertEqual(votes, [0, 0, 0, 0, 0])
+        self.assertEqual(len(votes), 5)
+
+    def test_set_votes(self):
+        # Make sure votes are set and saved.
+        self.product_page.votes.set_votes([1, 2, 3, 4, 5])
+        votes = self.product_page.votes.get_votes()
+        self.assertEqual(votes, [1, 2, 3, 4, 5])
+
+        # Ensure there's always 5 value set.
+        self.product_page.votes.set_votes([1, 2, 3, 4, 5, 6, 7])
+        self.assertEqual(votes, [1, 2, 3, 4, 5])
+        self.assertEqual(len(votes), 5)
+
+    def test_current_tally(self):
+        self.product_page.votes.set_votes([5, 4, 3, 2, 1])
+        current_tally = self.product_page.current_tally
+        self.assertEqual(current_tally, 15)
+
+        self.product_page.votes.set_votes([5, 5, 5, 5, 5])
+        current_tally = self.product_page.current_tally
+        self.assertEqual(current_tally, 25)
+
+    def test_creepiness(self):
+        self.product_page.current_vote_count = 100
+        self.product_page.votes.set_votes([5, 5, 5, 5, 5])
+        creepiness = self.product_page.creepiness
+        self.assertEqual(creepiness, 4)
+
+        self.product_page.current_vote_count = 0
+        self.product_page.votes.set_votes([0, 0, 0, 0, 0])
+        creepiness = self.product_page.creepiness
+        self.assertEqual(creepiness, 0)
+
+    def test_get_voting_json(self):
+        self.product_page.current_vote_count = 60
+        self.product_page.votes.set_votes([1, 2, 3, 4, 5])
```

Comment:
I think something may have gone wrong with the ProductPage field naming - `current_vote_count` is defined as:
```python
current_vote_count = models.IntegerField(default=0)  # The total points for creepiness
```
but the number of votes cast is already (correctly) determined by `current_tally`. 

We should rename `current_vote_count` to `current_total` or perhaps even more to the point of the data, `creepiness_value`.

Reply:
Yess good catch! This is definitely awkward naming. I'll rename it to `creepiness_value` so its more explicitly named. Thanks for catching this Pomax